pkg/apis/integreatly/v1alpha1/types.go

```diff
@@ -13,9 +19,13 @@ type ResourceTypeSpec struct {
 	SecretRef *SecretRef `json:"secretRef"`
 }
 
+type StatusPhase string
+
 // ResourceTypeStatus Represents the basic status information provided by a resource provider
 type ResourceTypeStatus struct {
-	Strategy  string     `json:"strategy,omitempty"`
-	Provider  string     `json:"provider,omitempty"`
-	SecretRef *SecretRef `json:"secretRef,omitempty"`
+	Strategy      string      `json:"strategy,omitempty"`
+	Provider      string      `json:"provider,omitempty"`
+	SecretRef     *SecretRef  `json:"secretRef"`
+	StatusPhase   StatusPhase `json:"statusPhase,omitempty"`
+	StatusMessage string      `json:"statusMessage,omitempty"`
```

Comment:
We can probably leave out the `status` prefix here, as the keys are in the `status` block

Reply:
Cool